src/storage/cookie-storage.js

```diff
@@ -1,16 +1,18 @@
 import {
   objectExtend,
   formatCookie,
+  getCookieDomain,
   parseCookies
 } from '../utils.js';
 
 class CookieStorage {
   constructor(defaultOptions) {
     this._defaultOptions = objectExtend({
-      domain: window.location.hostname,
+      domain: getCookieDomain(),
       expires: null,
       path: '/',
-      secure: false
+      secure: false,
+      getCookieFn: this._getCookie,
```

Comment:
I didn't have time to think about this logic these past few days, but my first impression is that I don't like it too much. The main reason is that the ability to authenticate a user on the server side is by using cookies. I will have to think a bit more about this.

For now, I suggest that you make it compilable in SSR, and later we can try to think about a solution to persist authentication state over SSR.